packages/manager/apps/procedures/src/data/api/proceduresApi.ts

```diff
@@ -37,18 +43,19 @@ export const getProceduresAPI = (procedure: Procedure) => {
       .then(({ data: dataResponse }) => dataResponse.uploadLinks);
   };
 
-  const finalize: () => Promise<void> = () => {
-    return v6.post(`${uri}/finalize`).then(({ data }) => data);
+  const finalize: <TFinalize>(finalizeData?: TFinalize) => Promise<void> = (
```

Comment:
I guess you set up this generic type to avoid having some `any`, but it feels unecessary as we won't manipulate this data and won't need to know in advance any of its property.
Also I think we'll always have a parameter of type `Record<string, any>` and could use this type as an alternative for the generic type